### ISSUE TITLE
V3 forced all RequestInfo not null 

### ISSUE THREAD

**Issue by cuong0993** [bkbnio/kompendium#411]
V2.3.5 allows nullable RequestInfo for Put, Patch and Post.
However, this change force all not null https://github.com/bkbnio/kompendium/pull/256.

From 

```
data class PutInfo<TParam, TReq, TResp>(
  val requestInfo: RequestInfo<TReq>?,
  override val responseInfo: ResponseInfo<TResp>,
  override val summary: String,
  override val description: String? = null,
  override val tags: Set<String> = emptySet(),
  override val deprecated: Boolean = false,
  override val securitySchemes: Set<String> = emptySet(),
  override val canThrow: Set<ExceptionInfo<*>> = emptySet(),
  override val parameterExamples: Set<ParameterExample> = emptySet(),
  override val operationId: String? = null
) : MethodInfo<TParam, TResp>
```
to ( newest code)

```
class PutInfo private constructor(
  override val request: RequestInfo,
  override val errors: MutableList<ResponseInfo>,
  override val response: ResponseInfo,
  override val tags: Set<String>,
  override val summary: String,
  override val description: String,
  override val externalDocumentation: ExternalDocumentation?,
  override val operationId: String?,
  override val deprecated: Boolean,
  override val parameters: List<Parameter>
) : MethodInfoWithRequest {

  companion object {
    fun builder(init: Builder.() -> Unit): PutInfo {
      val builder = Builder()
      builder.init()
      return builder.build()
    }
  }

  class Builder : MethodInfoWithRequest.Builder<PutInfo>() {
    override fun build() = PutInfo(
      request = request ?: error("request info must be present"),
      errors = errors,
      response = response ?: error("response info must be present"),
      tags = tags,
      summary = summary ?: error("Summary must be present"),
      description = description ?: error("Description must be present"),
      externalDocumentation = externalDocumentation,
      operationId = operationId,
      deprecated = deprecated,
      parameters = parameters
    )
  }
}
```

Can you help to make them nullable back? (A similar request here https://github.com/bkbnio/kompendium/issues/242)

**Comment by cuong0993** [comment]
And the description field too. I can see that https://redocly.github.io/redoc can parse a openapi.json with a request without requestBody and description.

**Comment by unredundant** [comment]
Does `requestType<Unit>()` not work?

**Comment by cuong0993** [comment]
Thank you. It works.

However, Insomnia error regarding the empty request. I think we can close this issue but it would be better to clean up redundant things.
![image](https://user-images.githubusercontent.com/8237154/215764377-a5bcb47d-c3c6-4150-9535-3ec760d57c52.png)

**Comment by unredundant** [comment]
Hmm... and if you remove the `requestBody` block entirely, insomnia does not complain?  If so this should be a pretty easy thing to fix, to simply remove that block in the case that the type is `Unit`

**Comment by Kulikovpavel** [comment]
Could you remove block for Unit requestType as you suppose?
`requestType<Unit>()` is not working correctly now, with result as `"requestBody":{"description":"xxx","required":true}`
Which is certainly not what we want

**Comment by unredundant** [comment]
Done, will be released shortly in `3.13.0`.
